### PR TITLE
Organization of utkscan header files

### PR DESCRIPTION
**Pull request by spaulaus**
These headers were making the directory unreadable and impossible quickly find information. The files have been reorganized and the source files updated to reflect this change.

### CODE REVIEW

**Comment by ksmith0** [comment]
This isn't what I expected. I would have gone the other way and had the following:
utkscan
|-analyzers
| |-include
| |-source
|-core
| |-include
| |-source
...

**Comment by ksmith0** [comment]
Also, I dislike putting paths in the include statements as the movement of one header file requires touching everything that uses it. I prefer to let the compiler figure out where the files are.

**Comment by spaulaus** [comment]
Fair point (though moot with IDEs.....). I like the suggestion and will make the correction.

**Comment by spaulaus** [comment]
@ksmith0 How does this strike your fancy?

File: Scan/utkscan/CMakeLists.txt [hunk 1]
```diff
@@ -34,16 +30,65 @@ if(UTKSCAN_VERBOSE)
...
+#------------------------------------------------------------------------------
+
+#Add the includes from ScanLib to our tree
+include_directories(include ../ScanLib/include)
+
+#Add the local include directories to the build tree
+include_directories(include analyzers/include)
+include_directories(include core/include)
+include_directories(include experiment/include)
+include_directories(include processors/include)
```
Comment (ksmith0) [inline Scan/utkscan/CMakeLists.txt]: Why do you repeatedly add the include directory?
Does all the following source need all these includes?

**Review by ksmith0 — CHANGES_REQUESTED** [review]
I would have expected something like

``` cmake
#All utkscan source needs the following headers
include_directories(core/include)

add_subdirectory(core/source)

add_subdirectory(analyzers)
add_subdirectory(processors)
add_subdirectory(experiments)
```

And the CMakeLists.txt in analyzers etc would have

``` cmake
include_directories(include)
add_subdirectory(source)
```

File: Scan/utkscan/CMakeLists.txt (hunk 1 above)
Comment (spaulaus) [reply Scan/utkscan/CMakeLists.txt]: Not all of it, but it's such a mixed bag that it's easier than cherry picking which ones need it and which do not. Remember that we have a mess of circular dependencies here.

File: Scan/utkscan/CMakeLists.txt [hunk 2]
```diff
@@ -34,16 +30,65 @@ if(UTKSCAN_VERBOSE)
...
 #Check if GSL is installed
 if(USE_GSL)
     find_package(GSL REQUIRED)
     add_definitions("-D usegsl")
 endif(USE_GSL)
 
+#------------------------------------------------------------------------------
+
+#Add the includes from ScanLib to our tree
+include_directories(include ../ScanLib/include)
```
Comment (ksmith0) [inline Scan/utkscan/CMakeLists.txt]: This line shouldn't be here. the ScanLib/include is added in the parent CMakeLists.txt

File: Scan/utkscan/CMakeLists.txt (hunk 1 above)
Comment (ksmith0) [reply Scan/utkscan/CMakeLists.txt]: Ok I'll accept that, but you need to drop the repeated addition of the include folder.

File: Scan/utkscan/CMakeLists.txt (hunk 2 above)
Comment (spaulaus) [reply Scan/utkscan/CMakeLists.txt]: When removing this line, I get a compilation error 

```
[ 84%] Building CXX object Scan/utkscan/core/source/CMakeFiles/CoreObjects.dir/utkscan.cpp.o
In file included from /home/vincent/programs/sources/utk/paass/Scan/utkscan/core/include/UtkScanInterface.hpp:7:0,
                 from /home/vincent/programs/sources/utk/paass/Scan/utkscan/core/source/utkscan.cpp:4:
/opt/PixieSuite2/include/ScanInterface.hpp:127:24: error: ‘XiaData’ has not been declared
  virtual bool AddEvent(XiaData *event_){ return false; }
                        ^~~~~~~
make[3]: *** [Scan/utkscan/core/source/CMakeFiles/CoreObjects.dir/build.make:447: Scan/utkscan/core/source/CMakeFiles/CoreObjects.dir/utkscan.cpp.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/Makefile2:657: Scan/utkscan/core/source/CMakeFiles/CoreObjects.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
[ 84%] Built target ProcessorObjects
make[1]: *** [CMakeFiles/Makefile2:559: Scan/utkscan/CMakeFiles/utkscan.dir/rule] Error 2
make: *** [Makefile:266: utkscan] Error 2
```

**Comment by ksmith0** [comment]
I clearly do not know how to use this review process yet. The requested changes are removal of 

``` cmake
include_directories(include ../ScanLib/include)
```

and remove repeated `include` from

``` cmake
include_directories(include analyzers/include)
include_directories(include core/include)
include_directories(include experiment/include)
include_directories(include processors/include)
```

**Comment by spaulaus** [comment]
I think we did exactly the same things....

**Comment by ksmith0** [comment]
I see you resolved the `BUILD_UTKSCAN_TESTS` issue. I propose that before we merge this we (you) move the tests folder to live inside analyzers. Either analyzers/tests or analyzers/source/tests

**Review by ksmith0 — CHANGES_REQUESTED** [review]
Changes to include_directories is good. Propose we also move tests while we are at it, I imagine the same thing will happen with them in the future.

**Comment by spaulaus** [comment]
I can see some merit in this. Though, each of the directories will get their own tests folders eventually. I was trying to centrally locate the tests, but as you say with the includes/sources it may make more sense to colocate all of them together.

**Comment by ksmith0** [comment]
I agree that each one will eventually get a test folder. This is also already true for `Core/tests`. Is the plan to move tests or not?

**Comment by spaulaus** [comment]
None of the core files have tests yet. I apologize for not being clear. I agree with the move of tests from utkscan/tests to utkscan/analyzers/tests.

**Comment by ksmith0** [comment]
Yes they [do](https://github.com/pixie16/paass/tree/master/Core/tests), although it is more of a test program. (Some info about use in issue #70)

**Comment by spaulaus** [comment]
Oh sorry! We have two "Cores" there's a core for utkscan as well, which is what I assumed was the reference given that this thread is mostly about that!

**Comment by ksmith0** [comment]
Ahh yeah, that is confusing.

**Comment by spaulaus** [comment]
@ksmith0 The request changes have been made. The organization is a little tighter, and more expansible.

**Review by ksmith0 — APPROVED** [review]
Looks good. Squash and merge?

**Comment by spaulaus** [comment]
Do it.

**Pull request merged**